### DISCUSSION TITLE
Unhelpful error message on invalid file

### DISCUSSION THREAD

**Discussion by shepting** [FiloSottile/age#415] [UX reports]
I was recently using `age` in an automated fashion for some CI machines and received the following message:
```
age: error: failed to read header: parsing age header: failed to read intro: EOF
```

Unfortunately, since the error message didn't specify either:
1. the `filename` of the invalid file _or_ 
2. the first line of the file it was parsing
I spent 30 minutes investigating what was wrong with my private key on my machine without realizing it was the "encrypted" file that was failing to parse!

My command was something like `cat ~/setup/id_rsa.encrypted | age --decrypt --identity ~/.age/ci.key --output ~/.ssh/id_rsa` so specifying that it was `STDIN` that was failing could help. Even moreso if it gave the first line of `STDIN` that didn't look right.

**Comment by FiloSottile** [comment]
That's definitely a bad error message, we'll work on it, thank you for reporting it!
